### PR TITLE
Fix generating terrain icon with certain image formats.

### LINKED ISSUE #84506: Image error when generating preview icon for non RGBA8 terrain tiles
### Godot version

5ee983188de97ae027f9b9c1443438063f708a7e

### System information

Windows 11

### Issue description

Godot editor will throw an error and fail to generate preview icons for terrain tiles which use a base image which is not RGBA8 format.
`core\io\image.cpp:2788 - Condition "format != p_src->format" is true.`

### Steps to reproduce

- Import a non-RGBA8 image into Godot.
- Create a TileMap
- Create an associated TileSet
- Create a terrain for this tileset
- Add the image as an atlas
- Go to the paint tab
- Paint the terrain on the TileSet
- An error will occur any time terrains are edited in the TileSet editor or selected in the TileMap editor terrain tab.

### Minimal reproduction project

[tileset_terrain_bug.zip](https://github.com/godotengine/godot/files/13264213/tileset_terrain_bug.zip)

### PR DESCRIPTION
**Pull request by SaracenOne**
Closes https://github.com/godotengine/godot/issues/84506

### CODE REVIEW

File: scene/resources/tile_set.cpp [hunk 1]
```diff
@@ -1839,20 +1839,26 @@ Vector<Vector<Ref<Texture2D>>> TileSet::generate_terrains_icons(Size2i p_size) {
...
+				ERR_FAIL_COND_V(src_texture.is_null(), output);
+				Ref<Image> src_image = src_texture->get_image();
+				ERR_FAIL_COND_V(src_image.is_null(), output);
 				Rect2i region = counts[terrain_set][terrain].region;
-				image->initialize_data(region.size.x, region.size.y, false, Image::FORMAT_RGBA8);
-				image->blit_rect(texture->get_image(), region, Point2i());
-				image->resize(p_size.x, p_size.y, Image::INTERPOLATE_NEAREST);
+
+				Image::Format src_image_format = src_image->get_format();
+				dst_image->initialize_data(region.size.x, region.size.y, false, src_image_format);
```
Comment (KoBeWi) [inline scene/resources/tile_set.cpp:1853]: Why not just
```suggestion
				dst_image->initialize_data(region.size.x, region.size.y, false, src_image->get_format());
```

**Comment by KoBeWi** [comment]
`texture->get_image()` does not return a reference to the texture's image, but it's copy. Thus it's redundant to have 2 Image variables.

**Comment by SaracenOne** [comment]
@KoBeWi Well, I removed the redunant extra format variable, but I'm looking over the code again and I'm confused what you mean in regards to the image variables.

**Comment by KoBeWi** [comment]
Ok disregard my comment. I missed that `src_image` is used twice, otherwise you could just reuse the `dst_image` ref.

**Review by KoBeWi — APPROVED** [review]

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.1.

**Pull request merged**